stingray/fourier.py

```diff
@@ -0,0 +1,1286 @@
+    ----------
+    times : float `np.array`
+        Array of times
+    gti : [[gti00, gti01], [gti10, gti11], ...]
+        good time intervals
+    segment_size : float
+        length of segments
+
+    Other parameters
+    ----------------
+    N : int, default None
+        Number of bins to divide the ``segment_size`` in
+    counts : float `np.array`, default None
+        Array of counts per bin
+
+    Returns
+    -------
+    cts : `np.array`
+        Array of counts
+    err : `np.array`
+        (optional) if ``errors`` is None, an array of errors in the segment
+
+    """
+    if counts is None and N is None:
+        raise ValueError(
+            "At least one between counts (if light curve) and N (if events) has to be set"
+        )
+
+    dt = None
+    if counts is not None:
+        dt = np.median(np.diff(times[:100]))
+
+    fun = _which_segment_idx_fun(counts, dt)
+
+    for s, e, idx0, idx1 in fun(times, gti, segment_size):
+        if idx1 - idx0 < 2:
+            yield None
+            continue
+        if counts is None:
+            event_times = times[idx0:idx1]
+            # counts, _ = np.histogram(event_times - s, bins=bins)
+            # astype here serves to avoid integer rounding issues in Windows,
+            # where long is a 32-bit integer.
+            cts = histogram((event_times - s).astype(float), bins=N,
+                            range=[0, segment_size]).astype(float)
+            cts = np.array(cts)
+        else:
+            cts = counts[idx0:idx1].astype(float)
+            if errors is not None:
+                cts = cts, errors[idx0:idx1]
+
+        yield cts
+
+
+def avg_pds_from_iterable(flux_iterable, dt, norm="abs", use_common_mean=True, silent=False):
+    """Calculate the average periodogram from an iterable of light curves
+
+    Parameters
+    ----------
+    flux_iterable : `iterable` of `np.array`s or of tuples (`np.array`, `np.array`)
+        Iterable providing either equal segments of light curve, or of light curve
+        and errors. They must all be of the same length.
+    dt : float
+        Time resolution of the light curves used to produce periodograms
+
+    Other Parameters
+    ----------------
+    norm : str, default "abs"
+        The normalization of the periodogram. "abs" is absolute rms, "frac" is
+        fractional rms, "leahy" is Leahy+83 normalization, and "none" is the
+        unnormalized periodogram
+    use_common_mean : bool, default True
+        The mean of the light curve can be estimated in each interval, or on
+        the full light curve. This gives different results (Alston+2013).
+        Here we assume the mean is calculated on the full light curve, but
+        the user can set ``use_common_mean`` to False to calculate it on a
+        per-segment basis.
+    silent : bool, default False
+        Silence the progress bars
+
+    Returns
+    -------
+    freq : `np.array`
+        The periodogram frequencies
+    pds : `np.array`
```

Comment:
I really think we should strive to be consistent with what we call things across stingray. I think we call this `power` everywhere else?

Reply:
It's actually all wrong. The return value is actually a Table, and this column is indeed called power.